easybuild/tools/filetools.py

```diff
@@ -2435,8 +2437,42 @@ def copy_file(path, target_path, force_in_dry_run=False):
             else:
                 mkdir(os.path.dirname(target_path), parents=True)
                 if path_exists:
-                    shutil.copy2(path, target_path)
-                    _log.info("%s copied to %s", path, target_path)
+                    try:
+                        # on filesystems that support extended file attributes, copying read-only files with
+                        # shutil.copy2() will give a PermissionError, when using Python < 3.7
+                        # see https://bugs.python.org/issue24538
+                        shutil.copy2(path, target_path)
+                        _log.info("%s copied to %s", path, target_path)
+                    except OSError as err:
+                        # catch the more general OSError instead of PermissionError, since python2 doesn't support
+                        # PermissionError
+                        if not os.stat(path).st_mode & stat.S_IWUSR:
+                            # failure not due to read-only file
+                            raise EasyBuildError("Failed to copy file %s to %s: %s", path, target_path, err)
```

Comment:
This part doesn't make sense to me, I think it should be removed...

If we give up here if the source file is read-only, then why bother with the attempt to fix the issue when using Python 3.6 via `shutil._copyxattr` below?

Reply:
Discussed with @jfgrimm in Slack, the condition is wrong here, should be `if os.stat`...

Also, it makes more sense to me to check whether the *target* file is read-only or not?

Fixed in https://github.com/jfgrimm/easybuild-framework/pull/1